python/rikai/types/vision.py

```diff
@@ -163,3 +178,16 @@ def to_numpy(self) -> np.ndarray:
                 self._cached_data = np.asarray(pil_img)
         assert self._cached_data is not None
         return self._cached_data
+
+    def to_bytes(self) -> bytes:
+        """Return a byte array.
+
+        Note
+        ----
+        The caller should close the image.
+        https://pillow.readthedocs.io/en/stable/reference/open_files.html#image-lifecycle
+        """
+        img = PILImage.open(self.open())
```

Comment:
same here with closing resources.

```
with self.to_pil() as img:
    img_bytes = BytesIO()
    img.save(img_bytes, format='PNG')
    return img_bytes.getvalue()
```

why PNG format as default? This is probably something that we should cleanup later and parameterize the image format.

Reply:
just went this way when 'JPEG/JPG' failed but I see now that I needed to specify the format in [lowercase](https://pillow.readthedocs.io/en/5.1.x/handbook/image-file-formats.html#jpeg)

Reply:
the bytes will differ per image format right? Maybe add in format/kwargs like #178 
see PIL docs for details: https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.save